### ISSUE TITLE
Beacon only fires to single random master in HA mode

### ISSUE THREAD

**Issue by doesitblend** [saltstack/salt#49663]
### Description of Issue/Question
We are seeing an issue where certain beacons are only firing to a single random master. In particular this happens with the `inotify` and the `process` beacon. The problem occurs when the beacon is firing with at least one master down in an HA environment. 

@DmitryKuzmenko is already aware of this issue and some of the details.

What we expect to see is that the beacon would fire to all alive masters.

### Setup
Configure 3-4 masters and connect at least a single minion in the environment. The minions should be connected in HA mode to all masters.

### Steps to Reproduce Issue
1. Configure the `inotify` beacon on the minion to watch any file 
2. Stop the first master configured in the minion
2. Watch the master event bus on the remaining alive masters
3. Update the watched file on the minion to trigger the beacon
4. You should see the event come in only once to a random master
5. Re-triggering the beacon may send the event to a different master

### Versions Report
```
#  Minion Version
    Salt Version:
               Salt: 2018.3.2
     
    Dependency Versions:
               cffi: Not Installed
           cherrypy: Not Installed
           dateutil: Not Installed
          docker-py: Not Installed
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.7.2
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: Not Installed
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.5.6
       mysql-python: Not Installed
          pycparser: Not Installed
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 2.7.5 (default, Jul 13 2018, 13:06:57)
       python-gnupg: Not Installed
             PyYAML: 3.11
              PyZMQ: 15.3.0
               RAET: Not Installed
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.2.1
                ZMQ: 4.1.4
     
    System Versions:
               dist: centos 7.5.1804 Core
             locale: UTF-8
            machine: x86_64
            release: 4.9.93-linuxkit-aufs
             system: Linux
            version: CentOS Linux 7.5.1804 Core
```

**Comment by garethgreenaway** [comment]
I believe the way it currently works is by design.  If you have the beacon fire to all masters then you're going to have all the masters potentially reacting to the event, which could end up with duplicate work.

**Comment by rickh563** [comment]
ZD-2774

**Comment by DmitryKuzmenko** [comment]
Fixed by #54247
